cdc/http_api_handler.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cdc
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/model"
+	cerror "github.com/pingcap/ticdc/pkg/errors"
+	"github.com/pingcap/ticdc/pkg/httputil"
+	"github.com/pingcap/tidb/store/tikv/oracle"
+	"go.uber.org/zap"
+)
+
+const (
+	// APIOpVarChangefeeds is the key of list option in HTTP API
+	APIOpVarChangefeeds = "state"
+)
+
+// JSONTime used to wrap time into json format
+type JSONTime time.Time
+
+// MarshalJSON use to specify the time format
+func (t JSONTime) MarshalJSON() ([]byte, error) {
+	stamp := fmt.Sprintf("\"%s\"", time.Time(t).Format("2006-01-02 15:04:05.000"))
+	return []byte(stamp), nil
+}
+
+// err of cdc http api
+type httpError struct {
+	Message string              `json:"error"`
+	Code    errors.RFCErrorCode `json:"errorCode"`
+}
+
+// ChangefeedCommonInfo holds some common usage information of a changefeed and use by RESTful API only.
+type ChangefeedCommonInfo struct {
+	ID             string              `json:"id"`
+	FeedState      string              `json:"state"`
+	CheckpointTSO  uint64              `json:"checkpoint-tso"`
+	CheckpointTime JSONTime            `json:"checkpoint-time"`
+	RunningError   *model.RunningError `json:"error"`
+}
+
+// handleHealth check if is this server is health
+func (s *Server) handleHealth(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+}
+
+// handleChangefeeds dispatch the request to the specified handleFunc according to the request method.
+func (s *Server) handleChangefeeds(w http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodGet {
+		s.handleChangefeedsList(w, req)
+		return
+	}
```

Comment:
what if http request method is not `GET`?

Reply:
It wiil return an ErrSupportGetOnly error now.